File: js_src/model/tavern.js

goog.provide('low.model.Tavern');



/**
 * The model for a player's tavern.
 * @param {number} gold
 * @constructor
 */
low.model.Tavern = function(gold) {

  /** @private {number} */
  this.gold_ = gold;
};


/** @return {number} */
low.model.Tavern.prototype.getGold = function() {
  return this.gold_;
};


/**
 * @param {!Object} json The JSON for a player's tavern object.
 * @return {!low.model.Tavern} The parsed tavern model.
 */
low.model.Tavern.fromJson = function(json) {
  return new low.model.Tavern(json['gold'] || 0);
};

File: src/low/model/Tavern.java
package low.model;

/**
 * Represents the resources in a player's tavern such as gold, fighters, etc.
 */
public class Tavern {
	
	private int gold;
	
	public Tavern() {
		gold = 0;
	}
	
	public int getGold() {
		return gold;
	}
	public void setGold(int gold) {
		this.gold = gold;
	}
}
